src/HtmlEntities.php

```diff
@@ -190,7 +58,6 @@ public function setDoubleQuote($doubleQuote)
      * If the value provided is non-scalar, the value will remain unfiltered
      *
      * @throws Exception\DomainException On encoding mismatches.
-     * @psalm-return ($value is scalar ? string : mixed)
      */
     public function filter(mixed $value): mixed
     {
```

Comment:
```suggestion
    {
        if (! is_scalar($value) || $value === '') {
```

Checking for an empty string here is defensive against later string length comparisons

Reply:
i changed the check also to `is_string` because otherwise a numeric value would be casted to an string